analyses/molecular-subtyping-EPN/02_ependymoma_generate_all_data.py

```diff
@@ -114,7 +114,8 @@ def broad_CNA_fill_df(row, CNA, arm, loss_gain):
 def fill_df_with_fpkm_zscores(df, fpkmdf, gene_name):
```

Comment:
I am concerned here that because `df` and `fpkmdf` are not necessarily the same size, they may also be out of order at times. Is there a check to be sure that the rows that satisfy `df["Kids_First_Biospecimen_ID_RNA"].notna()` have the same ids as `fpkmdf`?

Reply:
hmm there is no checks for the order. I can look into it

Reply:
I checked fpkm_df.loc["RELA",df.loc[df["Kids_First_Biospecimen_ID_RNA"].notna(),"Kids_First_Biospecimen_ID_RNA"]] which seems to have the same order as `df.loc[df["Kids_First_Biospecimen_ID_RNA"].notna(),"Kids_First_Biospecimen_ID_RNA"]` which is how I envision the order to be in line 115 in 02_ependymoma_generate_all_data.py. Is there a. better way to add a check in the code?

BS_4T0HPZDC     67.52
BS_H4NXBD2D    140.89
BS_S0175QKX     39.55
BS_KXMYBQ6N     38.31
BS_0BXY0F9N     26.66
                ...  
BS_C05FKR8X    162.01
BS_HZ707C55     41.73
BS_YE1MAQYJ     28.29
BS_PJPGVD62     79.53
BS_7B89YR6K     28.72

Reply:
Oh, I missed that on my first read: the fact that you are using `x["Kids_First_Biospecimen_ID_RNA"]` inside your lambda function means that you will get the correct values in every case.